tests/Feature/Appointments/StoreAppointmentTest.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Feature\Appointments;
+
+use Database\Factories\AppointmentFactory;
+use Database\Factories\ClinicFactory;
+use Database\Factories\DoctorFactory;
+use Database\Factories\UserFactory;
+use Lightit\Appointments\App\Exceptions\InvalidDatesException;
+use Lightit\Appointments\App\Exceptions\OverlappingException;
+use Lightit\Appointments\App\Exceptions\RelationException;
+use Lightit\Appointments\Domain\Actions\UpsertAppointmentAction;
+use Lightit\Appointments\Domain\Actions\ValidateClinicDoctorRelation;
+use Lightit\Appointments\Domain\Actions\ValidateDoctorOverlapping;
+use Lightit\Appointments\Domain\Actions\ValidateUserOverlapping;
+use Lightit\Appointments\Domain\DataTransferObjects\AppointmentDto;
+use Lightit\Appointments\Domain\Models\Appointment;
+
+use function Pest\Laravel\assertDatabaseHas;
+
+describe('appointments', function (): void {
+    it('throw an exception if the doctor already has an appointment', function (): void {
+        $doctor = DoctorFactory::new()->createOne();
+        $user = UserFactory::new()->createOne();
+        $clinic = ClinicFactory::new()->createOne();
+        $clinic->doctors()->attach($doctor);
+
+        $existingAppointment = AppointmentFactory::new()->create([
```

Comment:
I think you could use the factory as this

```suggestion
        $existingAppointment = AppointmentFactory::new()->for($user)->for($doctor)->for($clinic)->createOne()
```

Try to see if it works but I think it clears up the code